### DISCUSSION TITLE
Tautomer enumeration removes stereochemistry depending on how aromatic rings are defined in SMILES

### DISCUSSION THREAD

**Discussion by gedeck** [rdkit/rdkit#5397] [Q&A] [answered]
Before I file an issue, I would like to make sure that it is one. The Tautomer enumerator behaves differently if the structure is loaded from a SMILES where the aromatic bonds are defined using kekule format or not.
```
from rdkit import Chem
from rdkit.Chem.MolStandardize import rdMolStandardize
from rdkit.Chem import Draw

enumerator = rdMolStandardize.TautomerEnumerator()

smiles = 'c1ccc([C@@H](CC2=NCCN2)c2ccccn2)cc1'
mol = Chem.MolFromSmiles(smiles)
Draw.MolsToGridImage([mol, enumerator.Canonicalize(mol)])
```
In this case, the stereochemistry of the carbon is preserved:

![image](https://user-images.githubusercontent.com/8720575/176784310-ec5c88fa-156a-4dbd-a6df-59ecf9c7ad2b.png)

If on the other hand, the aromatic rings are described with single and double bonds
```
smiles = 'C([C@H](C1=CC=CC=C1)C2=NC=CC=C2)C3=NCCN3'
mol = Chem.MolFromSmiles(smiles)
Draw.MolsToGridImage([mol, enumerator.Canonicalize(mol)])
```
the stereochemistry is removed.
![image](https://user-images.githubusercontent.com/8720575/176784328-44bdaa56-d3fd-4ace-a85a-0ce64f679aab.png)

**Comment by greglandrum** [comment]
This looks like a bug to me. I'm away from my computer until Monday, so I can't confirm that though.

**Comment by greglandrum** [comment, marked as answer]
At first I thought this might be an order dependence (which obviously also shouldn't happen), but it's not:
```
In [12]: m = Chem.MolFromSmiles("c1ccc([C@@H](CC2=NCCN2)c2ccccn2)cc1")

In [13]: nm = Chem.Mol(m)

In [14]: Chem.Kekulize(nm, True)

In [15]: Chem.MolToSmiles(enum.Canonicalize(m))
Out[15]: 'c1ccc([C@@H](CC2=NCCN2)c2ccccn2)cc1'

In [16]: Chem.MolToSmiles(enum.Canonicalize(nm))
Out[16]: 'c1ccc(C(CC2=NCCN2)c2ccccn2)cc1'
```

**Comment by greglandrum** [comment]
Ok, my earlier diagnosis of this was wrong. The difference is actually due to the two structures kekulizing to different forms (which is an order dependence)... this leads to different tautomerization results.

Here's a gist illustrating the problem: https://gist.github.com/greglandrum/93df97219535ae1c4cade96038ee7140
